drivers/usb/udc/udc_bflb_bl61x.c

```diff
@@ -0,0 +1,1480 @@
+	}
+	if (enabled) {
+		tmp |= (1U << ((fifo_idx - 1) * USB_BL61X_FX_X_OFFSET + USB_EN_F0));
+	} else {
+		tmp &= ~(1U << ((fifo_idx - 1) * USB_BL61X_FX_X_OFFSET + USB_EN_F0));
+	}
+	sys_write32(tmp, cfg->base + USB_DEV_FCFG_OFFSET);
+}
+
+static void udc_bflb_bl61x_ep_set_out_mps(const struct device *dev,
+					uint8_t ep_idx,
+					uint16_t ep_mps)
+{
+	const struct udc_bflb_bl61x_config *cfg = dev->config;
+	uint32_t tmp;
+
+	tmp = sys_read32(cfg->base + USB_DEV_OUTMPS1_OFFSET
+		+ (ep_idx - 1) * USB_BL61X_XPS_X_OFFSET);
+	tmp &= ~USB_MAXPS_OEP1_MASK;
+	tmp |= ep_mps;
+	sys_write32(tmp, cfg->base + USB_DEV_OUTMPS1_OFFSET
+		+ (ep_idx - 1) * USB_BL61X_XPS_X_OFFSET);
+}
+
+static void udc_bflb_bl61x_ep_set_in_mps(const struct device *dev,
+					uint8_t ep_idx,
+					uint16_t ep_mps)
+{
+	const struct udc_bflb_bl61x_config *cfg = dev->config;
+	uint32_t tmp;
+
+	tmp = sys_read32(cfg->base + USB_DEV_INMPS1_OFFSET
+		+ (ep_idx - 1) * USB_BL61X_XPS_X_OFFSET);
+	tmp &= ~USB_MAXPS_IEP1_MASK;
+	tmp |= ep_mps;
+	tmp &= ~USB_TX_NUM_HBW_IEP1_MASK;
+	sys_write32(tmp, cfg->base + USB_DEV_INMPS1_OFFSET
+		+ (ep_idx - 1) * USB_BL61X_XPS_X_OFFSET);
+}
+
+static void udc_bflb_bl61x_fifo_reset_ctrl(const struct device *dev)
+{
+	const struct udc_bflb_bl61x_config *cfg = dev->config;
+	uint32_t tmp;
+
+	tmp = sys_read32(cfg->base + USB_DEV_CXCFE_OFFSET);
+	tmp |= USB_CX_CLR;
+	sys_write32(tmp, cfg->base + USB_DEV_CXCFE_OFFSET);
+}
+
+static void udc_bflb_bl61x_fifo_reset(const struct device *dev, uint8_t fifo_idx)
+{
+	const struct udc_bflb_bl61x_config *cfg = dev->config;
+	uint32_t tmp;
+
+	tmp = sys_read32(cfg->base + USB_DEV_FIBC0_OFFSET + 4 * (fifo_idx - 1));
+	tmp |= USB_FFRST0_HOV;
+	sys_write32(tmp, cfg->base + USB_DEV_FIBC0_OFFSET + 4 * (fifo_idx - 1));
+}
+
+/* fifo_idx : 1-4, ep_idx: 1-4
+ * ep_direction: 0 in 1 out
+ */
+static void udc_bflb_bl61x_ep_setfifo(const struct device *dev,
+					uint8_t ep_idx,
+					uint8_t fifo_idx,
+					uint8_t ep_dir)
+{
+	const struct udc_bflb_bl61x_config *cfg = dev->config;
+	uint32_t tmp;
+
+	ep_dir = ep_dir * 4;
+
+	if (ep_idx < 5) {
+		tmp = sys_read32(cfg->base + USB_DEV_EPMAP0_OFFSET);
+		tmp &= ~(0xf << ((ep_idx - 1) * 8 + ep_dir));
+		tmp |= ((fifo_idx - 1) << ((ep_idx - 1) * 8 + ep_dir));
+		sys_write32(tmp, cfg->base + USB_DEV_EPMAP0_OFFSET);
+	} else {
+		tmp = sys_read32(cfg->base + USB_DEV_EPMAP1_OFFSET);
+		tmp &= ~(0xf << ((ep_idx - 5) * 8 + ep_dir));
+		tmp |= ((fifo_idx - 1) << ((ep_idx - 5) * 8 + ep_dir));
+		sys_write32(tmp, cfg->base + USB_DEV_EPMAP1_OFFSET);
+	}
+}
+
+/* fifo_idx : 1-4, ep_idx: 1-4
+ * ep_direction: 0 in 1 out
+ * fifo_direction: 0 out 1 in 2 bidirectional
+ */
+static void udc_bflb_bl61x_fifo_setep(const struct device *dev,
+					uint8_t ep_idx,
+					uint8_t fifo_idx,
+					uint8_t fifo_dir)
+{
+	const struct udc_bflb_bl61x_config *cfg = dev->config;
+	uint32_t tmp;
+
+	if (ep_idx > 4 || fifo_idx > 4 || fifo_dir > 2) {
+		return;
+	}
```

Comment:
In case this was for checking invalid configuration, since only used internally:
```suggestion
	__ASSERT_NO_MSG(ep_idx <= 4);
	__ASSERT_NO_MSG(fifo_idx <= 4);
	__ASSERT_NO_MSG(fifo_dir <= 2);
```

Reply:
done